pkg/internal/codegen/parse/index.go

```diff
@@ -208,6 +209,41 @@ func parseResourceTag(tag string) resourceTags {
 	return result
 }
 
+// parseResourceAnnotation parses the tags in a "+resource=" comment into a resourceTags struct.
+func parseResourceAnnotation(t *types.Type) (resourceTags, error) {
+	result := resourceTags{}
+	for _, comment := range t.CommentLines {
+		anno := general.GetAnnotation(comment, "kubebuilder:resource")
+		if len(anno) == 0 {
+			continue
+		}
```

Comment:
So if users have specified resource tag in multiple lines, we will pick the values from the last one and will not warn users about it. 

As per my understanding, resource annotation should only exists once per type, so if we encounter multiple, we should fail. Also, we can restructure this logic to extract resourceAnnotation once through a helper fn (that should return error if multiple annotations encountered) and then call the parsing only once ?

Reply:
Yeah, I will make those changes. Thanks for clarifying that information.

Reply:
As per @DirectXMan12 comments, I tried to reuse the code already present in the index.go file(in commented state).I found some issues when I uncomment this (https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/index.go#L76) code where it's using following two functions to extract resource annotation. (`parseResourceTag`(https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/index.go#L187) and `getResourceTag` (https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/internal/codegen/parse/index.go#L236)). 

These functions are not handling the base case where the users provide two `resource` annotations. Also, the logic for the key-value pair can be imported from `pkg/internal/general`. Hence, I thought of redesigning this code a little bit (no major changes though). Also, instead of removing those two functions (`parseResourceTag` and `getResourceTag`), I commented out just in case anyone wants to reuse it for some purpose.
Let me know if your feedback

Reply:
you shouldn't leave commented code in the file.  The fact that there was commented code here before is wrong.

Reply:
otherwise, the new code looks better.

Reply:
Done!